enhancements/subscription-content/subscription-injection.md

```diff
@@ -0,0 +1,368 @@
+    name: <secret name with entitlement keys, must be in openshift-config namespace>
+  yumRepository:
+    name: <configMap name with yum repo definition, must be in openshift-config namespace>
+  aggregateToClusterRoles: # cluster roles which the ClusterBundle RBAC should aggregate to.
+                           # This can override the operator default
+  - edit
+  - admin
+status:
+  conditions:
+  - type: Invalid
+    status: "False" # standard k8s conditions, should obey the "abnormal-true" convention
+  - ... # additional informational conditions
+  clusterRole: <name>-subscription-bundle # clusterRole is cluster-scoped
+  shares:
+  - name: <name>-subscription-entitlement # shares are cluster-scoped
+  - name: <name>-subscription-yumRepository
+```
+
+#### Operator API
+
+```yaml
+kind: Operator
+apiVersion: subscription.openshift.io/v1alpha1
+metadata:
+  name: cluster # canonical
+  namespace: openshift-subscription-injection
+spec:
+  defaultClusterBundles: # default cluster bundles installed by the operator
+    - name: rhel7
+      namespace: openshift-config
+      entitlement: etc-pki-entitlement
+      yumRepository: redhat-repo-rhel7
+    - name: rhel8
+      namespace: openshift-config
+      entitlement: etc-pki-entitlement
+      yumRepository: redhat-repo-rhel8
+  defaultAggregatedClusterRoles: # default cluster roles that the ClusterBundle RBAC aggregates to
+    - edit
+    - admin
+status:
+  conditions: # standard status conditions, can be added as needed.
+    - type: Available
+      status: "True"
+```
+
+### Risks and Mitigations
+
+**Risk:** Entitlement keys can leak outside of the `openshift-config` namespace to unauthorized
+workloads.
+
+_Mitigation:_ Access to the entitlement keys is gated by the cluster role generated for the
+projected resource shares associated with the bundle. Cluster admins are responsible for either
+
+1. Creating a `ClusterRoleBinding` for the service accounts that need access to the entitlement, OR
+2. Declaring cluster roles to aggregate the `ClusterBundle` to via the `aggregateToClusterRoles`
+   array, OR
+3. Configuring the operator to aggregate the generated `ClusterRole` to appropriate system roles by
+   default.
+
+**Risk:** Shares and cluster roles are orphaned if the subscription `ClusterBundle` is deleted.
+
+_Mitigation:_ Owner references are used to ensure proper cleanup. Note that `ClusterBundle` is
+cluster-scoped, so it can an create owner references on any object.
+
+## Design Details
+
+### Open Questions
+
+1. Should a subscription bundle only allow one reference for `entitlement` and `yumRepository`?
+   Can/should we allow multiple?
+2. Should we restrict entitilement info to live in `openshift-config`? Should we allow entitlements
+   and yum repo info to live in any namespace?
+3. Will this work for multi-tenant clusters? Should we consider APIs that are namespace-scoped?
+4. Should we inject a subscription bundle into every container by default? If so, which one?
```

Comment:
i would frame this as you should inject it into pods by default (and you'd inject it into all containers in the pod if so).